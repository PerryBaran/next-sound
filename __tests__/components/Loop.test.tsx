import { fireEvent, render, screen } from "@testing-library/react"
import Loop from "../../src/components/musicPlayer/loop/Loop"

describe("Loop", () => {
  const props = {
    handleLoop: jest.fn()
  }

  test("renders correctly when loop equals 'song'", () => {
    render(<Loop handleLoop={props.handleLoop} loop="song" />)

    expect(screen.getByRole("button")).toBeInTheDocument()
    expect(screen.getByAltText("loop song")).toBeInTheDocument()
  })

  test("renders correctly when loop equals ''", () => {
    render(<Loop handleLoop={props.handleLoop} loop="" />)

    expect(screen.getByRole("button")).toBeInTheDocument()
    expect(screen.getByAltText("loop")).toBeInTheDocument()
  })

  test("render corretly when loop equal 'playlist'", () => {
    render(<Loop handleLoop={props.handleLoop} loop="playlist" />)

    expect(screen.getByRole("button")).toBeInTheDocument()
    expect(screen.getByAltText("loop playlist")).toBeInTheDocument()
  })

  test("button click", () => {
    render(<Loop handleLoop={props.handleLoop} loop="" />)

    expect(props.handleLoop).not.toHaveBeenCalled()

    fireEvent.click(screen.getByRole("button"))

    expect(props.handleLoop).toHaveBeenCalledTimes(1)
  })

  describe("snapshots", () => {
    test("loop equal ''", () => {
      const { asFragment } = render(
        <Loop handleLoop={props.handleLoop} loop="" />
      )

      expect(asFragment()).toMatchSnapshot()
    })

    test("loop equal 'song'", () => {
      const { asFragment } = render(
        <Loop handleLoop={props.handleLoop} loop="song" />
      )

      expect(asFragment()).toMatchSnapshot()
    })

    test("loop equal 'playlist'", () => {
      const { asFragment } = render(
        <Loop handleLoop={props.handleLoop} loop="playlist" />
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
