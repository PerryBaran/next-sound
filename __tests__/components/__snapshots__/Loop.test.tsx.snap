// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loop snapshots loop equal '' 1`] = `
<DocumentFragment>
  <button
    class="container faded"
    type="button"
  >
    <img
      alt="loop"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
</DocumentFragment>
`;

exports[`Loop snapshots loop equal 'playlist' 1`] = `
<DocumentFragment>
  <button
    class="container null"
    type="button"
  >
    <img
      alt="loop playlist"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
</DocumentFragment>
`;

exports[`Loop snapshots loop equal 'song' 1`] = `
<DocumentFragment>
  <button
    class="container null"
    type="button"
  >
    <img
      alt="loop song"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
</DocumentFragment>
`;
